File: solutions/kotlin/etl/1/src/main/kotlin/ETL.kt
/**
 * @author lusinabrian on 31/05/17.
 */
object ETL{
    
    /**
     * Maps the old data to a new system
     * @param oldMap old map with ints as keys and lists as values
     * @return [Map] new map with characters as keys and ints as values
     * */
    fun transform(oldMap : Map<Int, List<Char>>) : Map<Char, Int>{
        return oldMap.flatMap { 
            it.value.map { 
                word -> Pair(word.toLowerCase(), it.key)
            } 
        }.toMap()
    }
}

File: solutions/javascript/etl/1/etl.js
var ETL = function(){};

ETL.prototype.transform = function (legacy) {
  var scrabble = {};
  for(var x in legacy){
    for(var i = 0; i < legacy[x].length;i++){
      scrabble[legacy[x][i].toLowerCase()] = +x;
    }
  }
  return scrabble;
};

module.exports = ETL